pkg/dao/public_repositories.go

```diff
@@ -47,3 +49,22 @@ func (p publicRepositoryDaoImpl) List() (error, []PublicRepository) {
 	}
 	return nil, publicRepos
 }
+
+func (p publicRepositoryDaoImpl) UpdateRepository(pubRepo PublicRepository) error {
```

Comment:
I'm not sure about this approach, but I think public repositories part of the code is gonna get refactored away anyway.

Reply:
+1 makes sense.